framework/include/cppmicroservices/detail/ServiceTracker.hpp

```diff
@@ -263,7 +263,42 @@ namespace cppmicroservices
                 auto l = t->Lock();
                 if (t->Size_unlocked() == 0)
                 {
-                    t->WaitFor(l, rel_time, [&t] { return t->Size_unlocked() > 0 || t->closed; });
+                    BundleContext a = d->context;
+                    if (!a)
+                    {
+                        throw std::invalid_argument("The bundle context cannot be null.");
```

Comment:
A few comments on this line:
1. codecov is indicating that there is no code coverage for this line. I recommend adding a test that exercises this code path.
2. This should be a [logic_error](https://en.cppreference.com/w/cpp/error/logic_error) instead of `invalid_argument`. This failure happens when some other thread prematurely stops the bundle, which I would argue is a logic error.
3. The public doxygen API documentation for `WaitForService` needs to be updated to tell users about this exception.

Reply:
1. I agree, I have added a test... That test will be changed in future commits when we either get rid of the `GetLogSink` or change that it throws because right now we are catching a runtime error from that instead of the logic_error that we would like it to be. 
2. I agree
3. ServiceTracker.h has been updated